Bottom: 6b828c6efa95d272086ffe0c23a6362f170cbf1f
Top:    066fd757a1f1b4c2ba58d6bc03c216d5a74ccddd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-23 18:21:22 +0000

Fix preview drawing broken by resetting "Gathering" mode after painting.

Seems that the pinout preview widget relied on the bug fixed in commit
0647d028e6c755715d2fcdbd0b4f2be73d36ae7f. In the expose handler of the
preview, We should be calling the hid_expose_callback() function rather
than DrawElement() directly.



---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index 66bc1a4..72ec525 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -153,6 +153,7 @@ pinout_set_data (GhidPinoutPreview * pinout, ElementType * element)
 static gboolean
 ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * event)
 {
+  extern HID ghid_hid;
   GhidPinoutPreview *pinout = GHID_PINOUT_PREVIEW (widget);
   GdkDrawable *save_drawable;
   double save_zoom;
@@ -194,7 +195,7 @@ ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * event)
   gdk_draw_rectangle (widget->window, gport->bg_gc, TRUE, 0, 0, da_w, da_h);
 
   /* call the drawing routine */
-  DrawElement (&pinout->element, 0);
+  hid_expose_callback (&ghid_hid, NULL, &pinout->element);
 
   gport->drawable = save_drawable;
   gport->zoom = save_zoom;
